tests/libs/estdlib/test_gen_tcp.erl

```diff
@@ -129,27 +150,82 @@ test_listen_connect_parameters() ->
     [] = [Error || Error <- Results, Error =/= ok],
     ok.
 
-test_listen_connect_parameters(ListenMode, ConnectMode, ListenActive, ConnectActive) ->
-    {ok, ListenSocket} = gen_tcp:listen(0, [ListenMode, {active, ListenActive}, {buffer, 32}]),
+test_listen_connect_parameters(
+    InetClientBackend, InetServerBackend, ListenMode, ConnectMode, ListenActive, ConnectActive
+) ->
+    etest:flush_msg_queue(),
+
+    case get_otp_version() of
+        Version when Version =:= atomvm orelse (is_integer(Version) andalso Version >= 24) ->
+            ServerBackendOption = [{inet_backend, InetServerBackend}],
+            ClientBackendOption = [{inet_backend, InetClientBackend}];
+        _ ->
+            ServerBackendOption = [],
+            ClientBackendOption = []
+    end,
+
+    io:format(
+        "GEN_TCP-TEST> ServerBackendOption=~p ClientBackendOption=~p ListenMode=~p ConnectMode=~p ListenActive=~p ConnectActive=~p~n",
+        [
+            ServerBackendOption,
+            ClientBackendOption,
+            ListenMode,
+            ConnectMode,
+            ListenActive,
+            ConnectActive
+        ]
+    ),
+
+    NumMessages = 10,
+
+    {ok, ListenSocket} = gen_tcp:listen(
+        0,
+        ServerBackendOption ++ [ListenMode, {active, ListenActive}, {buffer, 32}]
+    ),
     {ok, {_Address, Port}} = inet:sockname(ListenSocket),
 
     Self = self(),
     ServerPid = spawn(fun() ->
         Self ! {self(), ready},
-        Result = test_listen_connect_parameters_accept(ListenMode, ListenActive, ListenSocket),
+        Result = test_listen_connect_parameters_accept(
+            ListenMode, ListenActive, ListenSocket, NumMessages, Self
+        ),
         Self ! {self(), Result}
     end),
     receive
         {ServerPid, ready} ->
             ok
     end,
 
-    {ok, Socket} = gen_tcp:connect(localhost, Port, [ConnectMode, {active, ConnectActive}]),
-    ok = test_listen_connect_parameters_client_loop(Socket, ConnectMode, ConnectActive, 10),
+    {ok, Socket} = gen_tcp:connect(
+        {127, 0, 0, 1},
+        Port,
+        ClientBackendOption ++ [ConnectMode, {active, ConnectActive}]
+    ),
+    ok = test_listen_connect_parameters_client_loop(
+        Socket, ConnectMode, ConnectActive, NumMessages
+    ),
+
+    %% race condition in active receive; client might
+    %% close connection before service has consumed (and delivered)
+    %% all messages to active recipient process.  So we need
+    %% to wait until the server has actually processed all the
+    %% messages it is expected to.
+    receive
+        server_done ->
+            ok
+    end,
+    % timer:sleep(50),
+
+    % erlang:display({closing, client, socket}),
```

Comment:
let's remove debug code.

Reply:
Removed, as well as other locations in this module